EGO Forum Enhancement.user.js

```diff
@@ -522,6 +523,91 @@ function handleApplicationPage() {
     });
 }
 
+/**
+ * Adds a "Find Issue Reason" button on the user awards page
+ * @returns void
+ */
+function handleUserAwardPage() {
+    var username = document.querySelector('.p-title-value').textContent.substring(0, document.querySelector('.p-title-value').textContent.indexOf("'s Awards"));
+    var blocks = document.querySelector('.blocks');
+    Array.from(blocks.children).forEach(function (block) {
+        var awardContainer = block.querySelector('div > .userAwardsContainer');
+        Array.from(awardContainer.children).forEach(function (award) {
+            var contentDiv = award.querySelector('div');
+            if(contentDiv.classList.contains('showAsDeleted')) {
+                return;
+            }
+            
+            var awardImageClasses = contentDiv.querySelector('.contentRow-figure > span > img').classList[1];
+            var awardId = awardImageClasses.substring(awardImageClasses.indexOf('--') + 2);
+            createButton("/award-system/" + awardId + "/recent?username=" + username, "Find Issue Reason" , contentDiv.querySelector('.contentRow-main'), '_blank', true);
+        });
+    });
+}
+
+/**
+ * Adds a form to the recent awards page to find a user's award
+ * @returns void
+ */
+function handleRecentAwardPage() {
+    var url = window.location.href;
+    var awardId = url.substring(url.indexOf('/award-system/') + 14, url.indexOf('/recent'));
+    var form = document.createElement('form');
+    form.setAttribute('action', '/award-system/' + awardId + '/recent');
+    form.setAttribute('method', 'get');
+    form.setAttribute('class', 'block');
+    form.innerHTML = '<div class="block-container"><h3 class="block-minorHeader">Find User Award</h3><div class="block-body block-row"><input type="text" class="input" data-xf-init="auto-complete" data-single="true" name="username" data-autosubmit="true" maxlength="50" placeholder="Name…" autocomplete="off"></div></div>';
+    var pageWrapper = document.querySelector('.p-pageWrapper');
+    pageWrapper.insertBefore(form, pageWrapper.querySelector('.p-body'));
+}
+
+/**
+ * Directs the url to the page containing the award of the username that was supplied
+ * @returns void
+ */
+function handleFindAwardPage() {
+    var url = window.location.href;
+    var awardId = url.substring(url.indexOf('/award-system/') + 14, url.indexOf('/recent'));
+    var userToFind = url.substring(url.indexOf('username=') + 9);
```

Comment:
This would be better done with Regex

```suggestion
    var match = url.match(/^https:\/\/www\.edgegamers\.com\/award-system\/(\d+)\/recent\?username=(.+)$/)
    var awardId = match[1];
    var userToFind = match[2];
```